docs/rules/array-callback-return.md

```diff
@@ -70,21 +80,119 @@ var foo = Array.from(nodes, function(node) {
 });
 
 var bar = foo.map(node => node.getAttribute("id"));
+
+myArray.forEach(function(item) {
+    return;
+});
 ```
 
-## Options
+Examples of **correct** code for the `{ "allowImplicit": true }` option:
 
-This rule has an object option:
+```js
+/*eslint array-callback-return: ["error", { allowImplicit: true }]*/
+var undefAllTheThings = myArray.map(function(item) {
+    return;
+});
+```
 
-* `"allowImplicit": false` (default) When set to true, allows implicitly returning `undefined` with a `return` statement containing no expression.
+Examples of **incorrect** code for the `{ "checkForEach": true }` option:
 
-Examples of **correct** code for the `{ "allowImplicit": true }` option:
+```js
+/*eslint array-callback-return: ["error", { checkForEach: true }]*/
+
+myArray.forEach(function(item) {
+    return handleItem(item)
+});
+
+myArray.forEach(function(item) {
+    if (item < 0) {
+        return x;
+    }
+    handleItem(item);
+});
+```
+
+Examples of **correct** code for the `{ "checkForEach": true }` option:
 
 ```js
-/*eslint array-callback-return: ["error", { allowImplicit: true }]*/
+/*eslint array-callback-return: ["error", { checkForEach: true }]*/
```

Comment:
It would be very nice to have some `forEach` examples with arrow functions, especially with those that implicitly return a value.